tests/integration/golang/aim/namespace/flows/app_test.go

```diff
@@ -0,0 +1,280 @@
+	helpers.BaseTestSuite
+}
+
+func TestAppFlowTestSuite(t *testing.T) {
+	suite.Run(t, &AppFlowTestSuite{})
+}
+
+func (s *AppFlowTestSuite) TearDownTest() {
+	assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+}
+
+func (s *AppFlowTestSuite) Test_Ok() {
+	tests := []struct {
+		name           string
+		namespace1Code string
+		namespace2Code string
+	}{
+		{
+			name:           "TestCustomNamespaces",
+			namespace1Code: "namespace-1",
+			namespace2Code: "namespace-2",
+		},
+		{
+			name:           "TestExplicitDefaultAndCustomNamespaces",
+			namespace1Code: "default",
+			namespace2Code: "namespace-1",
+		},
+		{
+			name:           "TestImplicitDefaultAndCustomNamespaces",
+			namespace1Code: "",
+			namespace2Code: "namespace-1",
+		},
+	}
+
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			defer func() {
+				assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+			}()
+
+			// setup namespaces
+			for _, nsCode := range []string{"default", tt.namespace1Code, tt.namespace2Code} {
+				// ignore errors here since default exists on first run
+				//nolint:errcheck
+				_, _ = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+					Code:                nsCode,
+					DefaultExperimentID: common.GetPointer(int32(0)),
+				})
+			}
+
+			// run actual flow test over the test data.
+			s.testAppFlow(tt.namespace1Code, tt.namespace2Code)
+		})
+	}
+}
+
+func (s *AppFlowTestSuite) testAppFlow(
+	namespace1Code, namespace2Code string,
+) {
+	// create Apps
+	app1ID := s.createApp(namespace1Code, &request.CreateApp{
+		Type: "tf",
+		State: request.AppState{
+			"app-state-key": "app-state-value1",
+		},
+	})
+
+	app2ID := s.createApp(namespace2Code, &request.CreateApp{
+		Type: "mpi",
+		State: request.AppState{
+			"app-state-key": "app-state-value2",
+		},
+	})
+
+	// test `GET /apps` endpoint with namespace 1
+	resp := []response.App{}
+	require.Nil(
+		s.T(),
+		s.AIMClient().WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 1 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app1ID, resp[0].ID)
+
+	// test `GET /apps` endpoint with namespace 2
+	require.Nil(
+		s.T(),
+		s.AIMClient().WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace2Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 2 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app2ID, resp[0].ID)
+
+	// IDs from other namespace cannot be fetched, updated, or deleted
+	errResp := response.Error{}
+	client := s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
```

Comment:
the same regarding to `response` type. it is by default `json` so you can avoid it.

Reply:
will remove